public/app/features/variables/state/actions.ts

```diff
@@ -72,12 +74,38 @@ export const initDashboardTemplating = (list: VariableModel[]): ThunkResult<void
       dispatch(addVariable(toVariablePayload(model, { global: false, index: orderIndex++, model })));
     }
 
+    templateSrv.updateTimeRange(getTimeSrv().timeRange());
+
     for (let index = 0; index < getVariables(getState()).length; index++) {
       dispatch(addInitLock(toVariablePayload(getVariables(getState())[index])));
     }
   };
 };
 
+export const completeDashboardTemplating = (dashboard: DashboardModel): ThunkResult<void> => {
+  return (dispatch, getState) => {
+    templateSrv.updateIndex();
```

Comment:
Hmm, will this have any effect as we haven't called templateSrv.init(variables)? We should have a short sync about this I think?

Reply:
hmm, this one can be skipped, I think. Good point! We don't need the index for the variabels. Only  for the globals and time.